From 185abd365c2898ad77dbb33a099341fa28e5a496 Mon Sep 17 00:00:00 2001
From: Kjell Ahlstedt <kjell.ahlstedt@bredband.net>
Date: Thu, 2 Jul 2015 19:53:16 +0200
Subject: 	Add #includes needed with glibmm newer than 2.31.0.

	* gdk/src/color.hg:
	* gdk/src/pixbufanimationiter.hg:
	* gdk/src/pixbufformat.hg:
	* gdk/src/rectangle.hg:
	* gdk/src/rgbcmap.hg:
	* gdk/src/types.hg:
	* gtk/src/enums.hg:
	* gtk/src/main.hg:
	* gtk/src/papersize.hg:
	* gtk/src/textattributes.hg:
	* gtk/src/treemodel.hg: Add individual includes now that gmmproc does not
	add #include <glibmm.h> at the top of every generated header.
	* gtk/gtkmm/border.h: Add #include <gtk/gtk.h> to make
	glibmm/tools/test_scripts/testheaders.sh happy.

diff --git a/ChangeLog b/ChangeLog
index 73e1eac..1d6d4d0 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,22 @@
+2015-07-02  Kjell Ahlstedt  <kjell.ahlstedt@bredband.net>
+
+	Add #includes needed with glibmm newer than 2.31.0.
+
+	* gdk/src/color.hg:
+	* gdk/src/pixbufanimationiter.hg:
+	* gdk/src/pixbufformat.hg:
+	* gdk/src/rectangle.hg:
+	* gdk/src/rgbcmap.hg:
+	* gdk/src/types.hg:
+	* gtk/src/enums.hg:
+	* gtk/src/main.hg:
+	* gtk/src/papersize.hg:
+	* gtk/src/textattributes.hg:
+	* gtk/src/treemodel.hg: Add individual includes now that gmmproc does not
+	add #include <glibmm.h> at the top of every generated header.
+	* gtk/gtkmm/border.h: Add #include <gtk/gtk.h> to make
+	glibmm/tools/test_scripts/testheaders.sh happy.
+
 2012-03-14  Murray Cumming  <murrayc@murrayc.com>
 
 	PaperSize: Correct the PaperSizeTraits pre-declaration.
diff --git a/gdk/src/color.hg b/gdk/src/color.hg
index ce22d4b..09c91fc 100644
--- a/gdk/src/color.hg
+++ b/gdk/src/color.hg
@@ -21,6 +21,8 @@
 // the #ifndef GDKMM_DISABLE_DEPRECATED in deprecated classes) is generated:
 _CONFIGINCLUDE(gdkmmconfig.h)
 
+#include <glibmm/arrayhandle.h>
+
 _DEFS(gdkmm,gdk)
 
 namespace Gdk
diff --git a/gdk/src/pixbufanimationiter.hg b/gdk/src/pixbufanimationiter.hg
index 4e12f52..f89a037 100644
--- a/gdk/src/pixbufanimationiter.hg
+++ b/gdk/src/pixbufanimationiter.hg
@@ -21,6 +21,7 @@
 
 #include <glibmm/object.h>
 #include <gdkmm/pixbuf.h>
+#include <glibmm/timeval.h>
 #include <gdk-pixbuf/gdk-pixbuf.h>
 
 _DEFS(gdkmm,gdk)
diff --git a/gdk/src/pixbufformat.hg b/gdk/src/pixbufformat.hg
index c460056..6c7daab 100644
--- a/gdk/src/pixbufformat.hg
+++ b/gdk/src/pixbufformat.hg
@@ -20,7 +20,7 @@
 _DEFS(gdkmm,gdk)
 
 #include <gdkmmconfig.h>
-
+#include <glibmm/arrayhandle.h>
 
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
 extern "C" { typedef struct _GdkPixbufFormat GdkPixbufFormat; }
diff --git a/gdk/src/rectangle.hg b/gdk/src/rectangle.hg
index f48de5a..01d1959 100644
--- a/gdk/src/rectangle.hg
+++ b/gdk/src/rectangle.hg
@@ -19,6 +19,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#include <glibmm/value.h>
 #include <gdk/gdk.h>
 
 _DEFS(gdkmm,gdk)
diff --git a/gdk/src/rgbcmap.hg b/gdk/src/rgbcmap.hg
index bfdfdfe..817661f 100644
--- a/gdk/src/rgbcmap.hg
+++ b/gdk/src/rgbcmap.hg
@@ -24,6 +24,7 @@ _DEFS(gdkmm,gdk)
 #m4 _POP()
 
 #include <gdkmmconfig.h>
+#include <glibmm/arrayhandle.h>
 
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
 extern "C" { typedef struct _GdkRgbCmap GdkRgbCmap; }
diff --git a/gdk/src/types.hg b/gdk/src/types.hg
index fd644b8..a05ba0c 100644
--- a/gdk/src/types.hg
+++ b/gdk/src/types.hg
@@ -19,6 +19,8 @@
 
 _DEFS(gdkmm,gdk)
 
+#include <glibmm/arrayhandle.h>
+#include <glibmm/listhandle.h>
 #include <gdk/gdk.h>
 #include <gdkmmconfig.h>
 
diff --git a/gtk/gtkmm/border.h b/gtk/gtkmm/border.h
index a64c669..b48017e 100644
--- a/gtk/gtkmm/border.h
+++ b/gtk/gtkmm/border.h
@@ -22,6 +22,7 @@
  */
 
 #include <glibmm/value.h>
+#include <gtk/gtk.h>  /* For GtkBorder */
 
 namespace Gtk
 {
diff --git a/gtk/src/enums.hg b/gtk/src/enums.hg
index 66c3696..ad3d20c 100644
--- a/gtk/src/enums.hg
+++ b/gtk/src/enums.hg
@@ -19,6 +19,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#include <glibmm/value.h>
 #include <gtkmmconfig.h> //For GTKMM_MAEMO_EXTENSIONS_ENABLED
 
 _DEFS(gtkmm,gtk)
diff --git a/gtk/src/main.hg b/gtk/src/main.hg
index c6c05ed..2665445 100644
--- a/gtk/src/main.hg
+++ b/gtk/src/main.hg
@@ -22,6 +22,7 @@
 
 #include <sigc++/sigc++.h>
 #include <gdkmm/types.h> //For GdkEventKey
+#include <glibmm/optioncontext.h>
 
 _DEFS(gtkmm,gtk)
 
diff --git a/gtk/src/papersize.hg b/gtk/src/papersize.hg
index 41fcb3c..3914cd1 100644
--- a/gtk/src/papersize.hg
+++ b/gtk/src/papersize.hg
@@ -17,6 +17,9 @@
 
 _DEFS(gtkmm,gtk)
 
+#include <glibmm/keyfile.h>
+#include <glibmm/listhandle.h>
+
 namespace Gtk
 {
 
diff --git a/gtk/src/textattributes.hg b/gtk/src/textattributes.hg
index b043f98..3966529 100644
--- a/gtk/src/textattributes.hg
+++ b/gtk/src/textattributes.hg
@@ -22,6 +22,8 @@
 
 _DEFS(gtkmm,gtk)
 
+#include <glibmm/value.h>
+
 namespace Gtk
 {
 
diff --git a/gtk/src/treemodel.hg b/gtk/src/treemodel.hg
index a04d924..360b4ed 100644
--- a/gtk/src/treemodel.hg
+++ b/gtk/src/treemodel.hg
@@ -18,6 +18,7 @@
  */
 
 #include <glibmm/interface.h>
+#include <glibmm/arrayhandle.h>
 #include <gtkmm/treeiter.h>
 
 _DEFS(gtkmm,gtk)
-- 
cgit v0.10.2

